packages/core/util/tests/Job-specs.js

```diff
@@ -131,4 +131,74 @@ describe('Job', function () {
 			j.idle('second');
 		});
 	});
+
+	describe('#promise', function () {
+		it('should throw when passed a non-thenable argument', function (done) {
+			const j = new Job(() => done('Unexpected job execution'));
```

Comment:
Should this have `new Error(...) in the done?

Reply:
It doesn't have to but probably should for consistency.

From https://mochajs.org/#asynchronous-code:
> This callback accepts both an Error instance (or subclass thereof) or a falsy value; anything else will cause a failed test.